contributors/design-proposals/bootstrap-discovery.md

```diff
@@ -173,6 +172,47 @@ In addition, `jws-kubeconfig-<token-id>-hash` will be set to the MD5 hash of the
 
 This controller will also delete tokens that are past their expiration time.
 
+## `kubeadm` UX
+
+We extend kubeadm with a set of flags and helper commands for managing and using these tokens.
+
+### `kubeadm init` flags
+
+* `--token` If set, this injects the bootstrap token to use when initializing the cluster.  If this is unset, then a random token is created and shown to the user.  If set explicitly to the empty string then no token is generated or created.  This token is used for both discovery and TLS bootstrap.
+* `--token-ttl` If set, this sets the TTL for the lifetime of this token.  Defaults to 0 which means "forever"
+
+### `kubeadm join` flags
+
+* `--token` This sets the token for both discovery and bootstrap auth.
+* `--discovery-url` If set this will grab the cluster-info data (a kubeconfig) from a URL. Due to the sensitive nature of this data, we will only support https URLs.  This also supports `username:password@host` syntax for doing HTTP auth.
+* `--discovery-file` If set, this will load the cluster-info from a file.
+* `--discovery-token` If set, (or set via `--token`) then we will be using the token scheme described above.
+
+Only one of `--discovery-url`, `--discovery-file` or `--discovery-token` can be set.  If more than one is set then an error is surfaced and `kubeadm join` exits.  Setting `--token` counts as setting `--discovery-token`.
+
+### `kubeadm token` commands
+
+`kubeadm` provides a set of utilities for manipulating token secrets in a running server.
+
+* `kubeadm token create` Creates a token server side.  With no options this'll create a token that is used for discovery and TLS bootstrap.
+  * `--token` The actual token value (in `id.secret` form) to write in.  If unset, a random value is generated.
+  * `--usages` A list of usages.  Defaults to `signing,authentication`.  Signing is used to sign the config-map used for discovery.  Authentication is used to authenticate for TLS bootstrap.
```

Comment:
Suggest: 
* `--usages` A list of usages. Defaults to `signing,authentication`.
   - If the `signing` usage is specified, the token will be used to JWS-sign the ConfigMap and can then be used for discovery.  
   - If the `authentication` usage is specified, the token can be used to authenticate for TLS bootstrap.